src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Shadow/PlatformShadowEffect.android.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Xamarin.Forms.Platform.Android;
+using Xamarin.Forms;
+using Android.Views;
+using AView = Android.Views.View;
+using Android.Content;
+using Android.OS;
+using System.ComponentModel;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.CommunityToolkit.Android.Effects;
+using Android.Widget;
+
+[assembly: ExportEffect(typeof(PlatformShadowEffect), nameof(ShadowEffect))]
+
+namespace Xamarin.CommunityToolkit.Android.Effects
+{
+	public class PlatformShadowEffect : PlatformEffect
+	{
+		const float defaultRadius = 10f;
+
+		const float defaultOpacity = 1f;
+
+		AView View => Control ?? Container;
+
+		protected override void OnAttached()
+		{
+			if (View == null)
+				return;
+
+			Update();
+		}
+
+		protected override void OnDetached()
+		{
+			if (View == null)
+				return;
+
+			View.SetElevation(0);
+		}
+
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+
+			if (View == null)
+				return;
+
+			switch (args.PropertyName)
+			{
+				case nameof(ShadowEffect.ColorPropertyName):
+				case nameof(ShadowEffect.OpacityPropertyName):
+				case nameof(ShadowEffect.RadiusPropertyName):
+				case nameof(ShadowEffect.OffsetXPropertyName):
+				case nameof(ShadowEffect.OffsetYPropertyName):
+					View.Invalidate();
+					Update();
+					break;
+			}
+		}
+
+		void Update()
+		{
+			if (Build.VERSION.SdkInt < BuildVersionCodes.Lollipop)
+				return;
+
+			var color = ShadowEffect.GetColor(Element);
+			if (color.IsDefault)
+			{
+				View.SetElevation(0);
+				return;
+			}
+
+			var radius = (float)ShadowEffect.GetRadius(Element);
+			if (radius < 0)
+				radius = defaultRadius;
+
+			var opacity = ShadowEffect.GetOpacity(Element);
+			if (opacity < 0)
+				opacity = defaultOpacity;
+
+			var androidColor = color.MultiplyAlpha(opacity).ToAndroid();
+
+			if (View is TextView textView)
+			{
+				var offsetX = (float)ShadowEffect.GetOffsetX(Element);
+				var offsetY = (float)ShadowEffect.GetOffsetY(Element);
+				textView.SetShadowLayer(radius, offsetX, offsetY, androidColor);
+				return;
+			}
+
+			View.SetElevation(View.Context.ToPixels(radius));
+
+			if (Build.VERSION.SdkInt < BuildVersionCodes.P)
```

Comment:
Does it make sense to support older versions with other alternatives (Example: create drawable, etc)?

Reply:
I am sure yes it is, but I don't know how to achieve it :) I would appreciate your help.
We can merge this one to develop (once I resolve other issues) and then enhance it (or maybe you can commit to this branch as well?)

Reply:
For now, colored shadows are supported for Labels (all apis) and all views (starting API 28)
Otherwise default black shadow will be shown